src-ts/tools/learn/free-code-camp/FreeCodeCamp.tsx

```diff
@@ -39,6 +47,29 @@ const FreeCodeCamp: FC<{}> = () => {
         { url: '/learn/fcc', name: lesson?.module.title ?? '' },
     ], [lesson])
 
+    const currentModuleData: LearnModule|undefined = useMemo(() => {
+        return courseData?.modules.find(d => d.key === moduleParam)
+    }, [courseData, moduleParam])
+
+    const currentStepIndex: number = useMemo(() => {
+      if (!currentModuleData) {
+          return 0
+      }
+
+      const lessonIndex: number = currentModuleData.lessons.findIndex(l => l.dashedName === lessonParam)
+      return lessonIndex + 1
+    }, [currentModuleData, lessonParam])
+
+    const handleNavigate: (direction: number) => void = useCallback((direction = 1) => {
+
+        const nextStep: LearnLesson|undefined = currentModuleData?.lessons[(currentStepIndex - 1) + direction]
+        if (!nextStep) {
+            return
+        }
+
+        navigate(`/learn/fcc?course=${courseParam}&module=${moduleParam}&lesson=${nextStep?.dashedName}`)
```

Comment:
can we move anything related to constructing routes, etc to the learn.routes file?